mod/caching/set/all/metric_answer_cache.rb

```diff
@@ -1,26 +1,39 @@
 # has to be called in finalize events so that in integrate stage all changes
 # are collected in @updated_meteric_answers and can be executed
-def refresh_answer_lookup_entry answer_id
+def update_answer id: nil, answer_id: nil, metric_id: nil
+  ids = if metric_id
+          Answer.where(metric_id: metric_id).pluck :id
+        elsif answer_id
+          Answer.where(answer_id: answer_id).pluck :id
+        else
+          [id]
+        end
```

Comment:
seems like this should be two or three methods, though it's not obvious to me how best to do it  maybe:
```
def update_answers *filters
  ids = answer_ids_to_update *filters
  update_answers_now_or_later ids
end
``` 
?